lib/goliath/request.rb

```diff
@@ -63,6 +63,8 @@ def initialize(app, conn, env)
         end
       end
 
+      @env[GOLIATH_REQUEST] = self
```

Comment:
Any memory leak concerns here? You're creating a self-reference, which can cause trouble..